### PR TITLE
[RNMobile][Monorepo] Fix linting

### PR DESCRIPTION
**Pull request by ceyhun**
## Description

This is a part of the migration of gutenberg-mobile to the gutenberg repo. See #18508

- Fixed all lint-js errors (mostly with `npm run lint-js:fix`)
- Fixed trivial warnings like having `/** @format */` or `/** @flow */` on top of files
- There are still some warnings about `Illegal usage of jasmine global eslint(jest/no-jasmine-globals)` that are not trivial and can be handled on a later stage
- Removed the remaining flow types

## How has this been tested?

- `npm run lint` should complete without errors
- add a `console.log` to a .js file in `packages/react-native-editor/` and `npm run lint` should return an error `Unexpected console statement  no-console`

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by Tug** [comment]
@ceyhun I'm seeing this error when running `npm run lint` locally:
```
[0] ../gutenberg/packages/react-native-editor/__device-tests__/helpers/utils.js
[0]   5:1  error  'wd' should be listed in the project's dependencies. Run 'npm i -S wd' to add it  import/no-extraneous-dependencies
[0]
```
Any idea?

**Comment by ceyhun** [comment]
@Tug It is correct that you're seeing that error and it should be fixed. I'm not sure why I'm not seeing it. I did a clean run again and I'm still not seeing it.

**Comment by Tug** [comment]
I know this is out of scope of this PR but I took the opportunity to run the unit tests. The main ones are currently failing with multiple syntax errors around `react-native-editor/__device-tests__`:
```
 Summary of all failing tests
 FAIL  packages/react-native-editor/__device-tests__/gutenberg-editor-paragraph.test.js
  ● Test suite failed to run

    SyntaxError: .../gutenberg/packages/react-native-editor/__device-tests__/pages/editor-page.js: Support for the experimental syntax 'classProperties' isn't currently enabled (13:21):

      11 |
      12 | export default class EditorPage {
    > 13 | 	paragraphBlockName = 'Paragraph';
...
```

It also seems jest is failing to interpret flow code from RN:
```
 FAIL  packages/react-native-editor/src/index.test.js
  ● Test suite failed to run

    .../gutenberg/node_modules/react-native/Libraries/Utilities/warnOnce.js:15
    const warnedKeys: {[string]: boolean} = {};
          ^^^^^^^^^^

    SyntaxError: Missing initializer in const declaration

      at ScriptTransformer._transformAndBuildScript (node_modules/@jest/transform/build/ScriptTransformer.js:537:17)
      at ScriptTransformer.transform (node_modules/@jest/transform/build/ScriptTransformer.js:579:25)
      at Object.<anonymous> (node_modules/react-native/Libraries/react-native/react-native-implementation.js:14:18)
```

We'll need to either update the main gutenberg jest config (in test/unit/jest.config.js) to use react-native preset or most probably exclude `packages/react-native-*` from the main tests

**Comment by Tug** [comment]
> @Tug It is correct that you're seeing that error and it should be fixed. I'm not sure why I'm not seeing it. I did a clean run again and I'm still not seeing it.

I'll keep trying and see if I can fix it on my side 👍

**Comment by ceyhun** [comment]
> We'll need to either update the main gutenberg jest config (in test/unit/jest.config.js) to use react-native preset or most probably exclude `packages/react-native-*` from the main tests

Ignored test files in `react-native-*` packages folders in main unit tests `test/unit/jest.config.js` but enabled them in main native tests `test/native/jest.config.js`.

- `npm run test-unit` 
- `npm run test-unit:native` 

should run without errors now, except 1 error in `block-serialization-spec-parser/test/index.js` and 1 in `storybook/test/index.js` but I think those are not related to the changes here.

File: test/unit/jest.config.js [hunk 1]
```diff
@@ -30,6 +30,7 @@ module.exports = {
 		'<rootDir>/.*/build/',
 		'<rootDir>/.*/build-module/',
 		'<rootDir>/.+.native.js$',
+		'packages/react-native-*',
```
Comment (Tug) [inline test/unit/jest.config.js:33]: Really minor but we could try to be a bit more consistent with `/packages/react-native-*`

Comment (ceyhun) [reply test/unit/jest.config.js]: Updated 👍

**Review by Tug — APPROVED** [review]
LGTM 👍 

As usual, let's have @hypest merge with a merge commit

**Comment by hypest** [comment]
Will merge with "Allow merge commits" ON and then disable it again.

**Pull request merged**